core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/promisehandlers/OpenOperatorHandler.scala

```diff
@@ -0,0 +1,18 @@
+package edu.uci.ics.amber.engine.architecture.worker.promisehandlers
+
+import edu.uci.ics.amber.engine.architecture.worker.WorkerAsyncRPCHandlerInitializer
+import edu.uci.ics.amber.engine.architecture.worker.promisehandlers.OpenOperatorHandler.OpenOperator
+import edu.uci.ics.amber.engine.common.rpc.AsyncRPCServer.ControlCommand
+
+object OpenOperatorHandler {
+
+  final case class OpenOperator() extends ControlCommand[Unit]
+}
+
+trait OpenOperatorHandler {
+  this: WorkerAsyncRPCHandlerInitializer =>
+
+  registerHandler { (openOperator: OpenOperator, sender) =>
+    operator.open()
```

Comment:
I think we need to add some checks here to prevent an operator from opening multiple times.

Reply:
semantically, can an operator be opened multiple times? I think the operator api does not disallow this. you can view the `OpenOperator ` as a wrapper for this operator api. 

also currently only controller are sending those `OpenOperator` commands, so I would against the idea to restrict single execution of open. 

what do you think?

Reply:
let's ask for @zuozhiw 's input.

Reply:
to add check on exactly once for `open`, we need to add flags in the `Operator` interface. similar things should be done for `close`, or even consider the situation for `re-open a closed operator`. I suggest we move this to later refactoring.

Reply:
created issue to discuss later. #1425